src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -715,4 +715,24 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_OFI_unregister_am_bufs(void)
 #undef CQ_D_HEAD
 #undef CQ_D_TAIL
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_gpu_malloc_pack_buffer(void **ptr, size_t pack_size)
+{
+    if (MPIDI_OFI_ENABLE_HMEM) {
+        return MPL_gpu_malloc_host(ptr, pack_size);
+    } else {
+        *ptr = MPL_malloc(pack_size, MPL_MEM_BUFFER);
+        return 0;
```

Comment:
What is the reason? Is it because ZE malloc'ed buffer cannot be handled by certain providers? Or is it ZE wants to use fast copy path? I think the logic should be handled at caller site rather than in the utility wrappers.

Reply:
Yeah it is because certain providers cannot handle ZE-malloced buffers

Reply:
I see. Do not do this here. Explicitly do the logic at the caller site -- ideally, guarded within the `HAVE_ZE` macros.

Reply:
This malloc is used for GPU pipeline as well. Is it okay to add the `HAVE_ZE` guards inside the method instead?

Reply:
@hzhou thoughts?

Reply:
Thanks for reminding me. I'll take a look later today.

Reply:
Added the macro guard and a comment in the utility wrapper since it is needed for GPU pipeline in a later PR

Reply:
I've ended up dropping the commit, because it may not be needed anymore. We will re-evaluate its use for GPU pipeline.